service/header/store.go

```diff
@@ -140,10 +130,21 @@ func (s *store) Stop(ctx context.Context) error {
 
 func (s *store) Head(ctx context.Context) (*ExtendedHeader, error) {
 	head, err := s.GetByHeight(ctx, s.heightSub.Height())
-	if err == ErrNotFound {
+	if err == nil {
```

Comment:
~~In the case `err != nil` the error might get overwritten by readHead and not handled a few lines below?~~
Ah, it's handled by the switch case.